Cant install package on OSX

Always getting this fault when I try to install the package. 

`Installing “arduino-upload@1.3.2” failed.Hide output…

> serialport-builds-electron@4.0.7 install /private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron
> node-pre-gyp install --fallback-to-build

Failed to execute '/Applications/Atom.app/Contents/Resources/app/apm/bin/node /Applications/Atom.app/Contents/Resources/app/apm/bin/../node_modules/.bin/node-gyp configure --fallback-to-build --module=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release/serialport.node --module_name=serialport --module_path=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release' (1)
/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9
`-- (empty)


node-pre-gyp ERR! Tried to download(500): https://files.sorunome.de/arduino-upload/serialport/4.0.7/serialport-v4.0.7-electron-v1.6-darwin-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for serialport-builds-electron@4.0.7 and electron@1.6.15 (electron-v1.6 ABI) (falling back to source compile with node-gyp) 
gyp ERR! configure error 
gyp ERR! stack Error: Python executable "/Applications/Atom.app/Contents/Resources/app/apm/bin/python-interceptor.sh" is v3.6.1, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at failPythonVersion (/Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:454:14)
gyp ERR! stack     at /Applications/Atom.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/configure.js:443:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:197:7)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
gyp ERR! System Darwin 16.7.0
gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "configure" "--fallback-to-build" "--module=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release/serialport.node" "--module_name=serialport" "--module_path=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release"
gyp ERR! cwd /private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/Applications/Atom.app/Contents/Resources/app/apm/bin/node /Applications/Atom.app/Contents/Resources/app/apm/bin/../node_modules/.bin/node-gyp configure --fallback-to-build --module=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release/serialport.node --module_name=serialport --module_path=/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron/build/Release' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:106:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:877:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
node-pre-gyp ERR! System Darwin 16.7.0
node-pre-gyp ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/node_modules/arduino-upload/node_modules/serialport-builds-electron
node-pre-gyp ERR! node -v v6.9.5
node-pre-gyp ERR! node-pre-gyp -v v0.6.38
node-pre-gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/package.json'
npm WARN apm-install-dir-117929-77429-uih4nw.272gdjkyb9 No description
npm WARN apm-install-dir-117929-77429-uih4nw.272gdjkyb9 No repository field.
npm WARN apm-install-dir-117929-77429-uih4nw.272gdjkyb9 No README data
npm WARN apm-install-dir-117929-77429-uih4nw.272gdjkyb9 No license field.
npm ERR! Darwin 16.7.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/User/.atom/.apm/.apmrc" "--userconfig" "/Users/User/.atom/.apmrc" "install" "/private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/d-117929-77429-3py533.xubv9m5cdi/package.tgz" "--runtime=electron" "--target=1.6.15" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! serialport-builds-electron@4.0.7 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the serialport-builds-electron@4.0.7 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the serialport-builds-electron package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs serialport-builds-electron
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls serialport-builds-electron
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/0y/4dl944691xxgnd8csdhp_ksm0000gn/T/apm-install-dir-117929-77429-uih4nw.272gdjkyb9/npm-debug.log
npm ERR! code 1
`

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,4 @@
 .DS_Store
 npm-debug.log
 node_modules
-screenshots
+package-lock.json

--- a/README.md
+++ b/README.md
@@ -15,7 +15,7 @@ Did you not like the look of the arduino IDE but were stuck with it nevertheless
 (Requires atom 1.10 or later)
 
 ## Troubleshooting
-* Are you on Windows and can't install because of the serialport package? Try installing the [Visual C++ 2015 building tools](https://github.com/felixrieseberg/windows-build-tools) by running `npm install --global --production windows-build-tools` from `%LocalAppData%\atom\app-<VERSION>\resources\apm\node_modules\npm\bin` (this uses atoms integrated npm instead of requiring an external instance). Then update apm config with `apm config set msvs_version [VERSION]` (the version installed by the tool should be 2015) and `apm config set python [PATH]\python.exe` (where PATH is the path to the python executable).
+* Are you on Windows and can't install because of the serialport package? Try installing the [Visual C++ 2015 building tools](https://github.com/felixrieseberg/windows-build-tools) by running `.\npm install --global --production windows-build-tools` from `%LocalAppData%\atom\app-[VERSION]\resources\app\apm\bin` in an adminastrative PowerShell (this uses atoms integrated npm instead of requiring an external instance). Then update apm config with `apm config set msvs_version [VERSION]` (the version installed by the tool should be 2015) and `apm config set python [PATH]\python.exe` (where PATH is the path to the python executable).
 * The verify/build output does not appear? Make sure your Arduino official IDE language is set to English.
 
 ## Available commands

--- a/lib/arduino-upload.coffee
+++ b/lib/arduino-upload.coffee
@@ -11,7 +11,7 @@ tmp = require 'tmp'
 Boards = require './boards'
 
 try
-	serialport = require 'serialport-builds-electron'
+	serialport = require 'serialport'
 catch e
 	serialport = null
 try
@@ -295,10 +295,10 @@ module.exports = ArduinoUpload =
 	serialNormalClose: true
 	_openserialport: (port, start = true)->
 		try
-			serial = new serialport.SerialPort port, {
-					baudRate: atom.config.get('arduino-upload.baudRate')
-					parser: serialport.parsers.raw
-				}
+			serialport serial = new serialport(port, {
+						baudRate: atom.config.get('arduino-upload.baudRate')
+						parser: serialport.parsers.raw
+					});
 			@serialNormalClose = true
 			serial.on 'open', (data) =>
 				if start

--- a/package.json
+++ b/package.json
@@ -13,22 +13,20 @@
   "repository": "https://github.com/Sorunome/arduino-upload",
   "license": "MIT",
   "engines": {
-    "atom": ">=1.0.0 <2.0.0"
+    "atom": ">=1.0.0"
   },
   "consumedServices": {
     "status-bar": {
       "versions": {
-        "^1.0.0": "consumeStatusBar"
+        ">=1.0.0": "consumeStatusBar"
       }
     }
   },
   "dependencies": {
     "atom-space-pen-views": ">=2.0.3",
-    "bluebird": "^3.5.0",
-    "serialport-builds-electron": "^4.0.7",
-    "tmp": ">=0.0.31"
-  },
-  "optionalDependencies": {
-    "usb-detection": ">=1.4.0"
+    "bluebird": ">=3.5.0",
+    "serialport": ">=7.0.2",
+    "tmp": ">=0.0.31",
+    "usb-detection": ">=3.0.0"
   }
 }

